### PR TITLE
20.3 fb panoramapublic resubmit

### PR DESCRIPTION
**Pull request by vagisha**
#### Rationale
[Issue: 40564](https://www.labkey.org/MacCoss/Issue%20Tracker/issues-details.view?issueId=40564) Automate copying an experiment that is re-submitted.  Some other fixes.

#### Changes
- Copy PX ID (if present) and assign the same reviewer to the new copy.  Rename the folder for the old copy as V.1, V.2 etc.  Delete old folder if that option is checked by admin doing the copy.
- Show missing details required for a PX submission when user re-submits data that has a PX ID.
- Validate data is complete for a PX submission (if PX ID was requested) before copying to Panorama Public.
- Skip ProteomeXchange checks for small molecule data. Cannot assign PX ID to small molecule only data.
- Do not allow re-submitting data that is already public and associated with a publication on Panorama Public. ([Issue 40565](https://www.labkey.org/MacCoss/Issue%20Tracker/issues-details.view?issueId=40565)).
- Check for matching .wiff and .wiff.scan files when doing raw data check.
- Updated test and added new test files.

### CODE REVIEW

File: panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java [hunk 1]
```diff
@@ -403,4 +421,67 @@ public static DataLicense getLicenseSelectedForSubmission(Integer submittedExper
...
+        {
+            // If this experiment has already been copied and the journal copy is final (paper published and data public)
+            // then the user should not be able to re-submit this data.
+            ExperimentAnnotations journalCopy = ExperimentAnnotationsManager.get(journalExperiment.getJournalExperimentId());
+            return journalCopy == null || !journalCopy.isFinal();
+        }
+        return true;
+    }
+
+    public static ExperimentAnnotations getJournalCopy(ExperimentAnnotations expAnnotations)
```
Comment (labkey-jeckels) [inline panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java]: Consider @Nullable

File: panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java [hunk 2]
```diff
@@ -56,9 +61,9 @@
 {
     private ExperimentAnnotationsManager() {}
 
-    public static ExperimentAnnotations get(int experimentAnnotationsId)
+    public static ExperimentAnnotations get(Integer experimentAnnotationsId)
```
Comment (labkey-jeckels) [inline panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java]: Consider @Nullable

File: panoramapublic/src/org/labkey/panoramapublic/query/JournalManager.java [hunk 3]
```diff
@@ -427,23 +429,25 @@ public static void removeJournalAccess(ExperimentAnnotations expAnnotations, Jou
...
-        // that references this URL we should not get a foreign key constraint error.
-        tryDeleteShortUrl(je.getShortCopyUrl(), user);
-        // Try to delete the short access URL only if the experiment has not yet been copied (accessURL points to journal's folder after copy)
-        // OR the access url is no longer referenced in the ExperimentAnnotations table.
-        if(je.getCopied() == null || ExperimentAnnotationsManager.getExperimentForShortUrl(je.getShortAccessUrl()) == null)
+        if(je.getJournalExperimentId() == null)
         {
-            tryDeleteShortUrl(je.getShortAccessUrl(), user);
+            // This experiment has not yet been copied to Panorama Public so we can delete the row in JournalExperiment
+            SimpleFilter filter = new SimpleFilter();
```
Comment (labkey-jeckels) [inline panoramapublic/src/org/labkey/panoramapublic/query/JournalManager.java:435]: Does any upstream code validate that the supplied IDs belong to a container that the user has permission to operate on?

**Review by labkey-jeckels — APPROVED** [review]
Couple of suggestions on adding @Nullable annotations, and one question on validating IDs are owned by a container with the right permissions

File: panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java (hunk 2 above)
Comment (vagisha) [reply panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java]: Done

File: panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java (hunk 1 above)
Comment (vagisha) [reply panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java]: Done

File: panoramapublic/src/org/labkey/panoramapublic/query/JournalManager.java (hunk 3 above)
Comment (vagisha) [reply panoramapublic/src/org/labkey/panoramapublic/query/JournalManager.java:435]: Yes. The action requires AdminPermission, and there is an ensureCorrectContainer() method that redirects to the container that the ID belongs to.   I did a quick review of all the actions in PanoramaPublicController and added the container check in two places where it wasn't being done.

**Pull request merged**